Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/data/source/InMemoryRemoteTaskDataSource.java

```diff
@@ -150,7 +150,25 @@ public void call(Task task) {
                 dataSourceTasks.remove(taskId);
                 return Observable.empty();
             }
-        }).delay(remoteDelay, remoteDelayUnit, Schedulers.immediate());
+        }).compose(this.<Void>delay());
+    }
+
+    private <T> Observable.Transformer<T, T> delay() {
```

Comment:
I tend to follow the rule of three, duplicate once refactor when you have 3 instances.
In this case it is a class mocking behaviour, I would probably leave it at that. If we have the need for a 3rd then we can refactor.